### PR TITLE
Remove Raptor legacy connector

### PR DESCRIPTION
**Pull request by mosabua**
## Description

The connector has no known use and forces some complexities on the rest of the codebase. Do allow next clean up steps the connector is removed.

In a contributor call, on slack, and in other conversations no concerns were expressed by anybody about the removal.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
## Raptor connector
* {{breaking}} Remove the Raptor connector. ({issue}`23588`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
YES!

**Review by wendigo — APPROVED** [review]

**Comment by mosabua** [comment]
Look at that .. all green. Not even failures from flaky, unrelated tests. I won the lotto...

**Comment by mosabua** [comment]
Can I get an approval from @dain and/or @electrum ?

**Comment by mosabua** [comment]
We requested user input on slack and linkedin to find out if anyone uses this plugin now.

File: lib/trino-orc/src/main/java/io/trino/orc/OutputStreamOrcDataSink.java [hunk 1]
```diff
@@ -41,15 +41,6 @@ public static OutputStreamOrcDataSink create(TrinoOutputFile outputFile)
         return new OutputStreamOrcDataSink(outputFile.create(memoryContext), memoryContext);
     }
 
-    // Do not use this method, it is here only for io.trino.plugin.raptor.legacy.storage.OrcFileWriter.createOrcDataSink
-    // and it should be removed in the future
-    @Deprecated
```
Comment (electrum) [inline lib/trino-orc/src/main/java/io/trino/orc/OutputStreamOrcDataSink.java:46]: Nice find

**Review by electrum — APPROVED** [review]

**Review by raunaqmorarka — APPROVED** [review]

**Comment by wendigo** [comment]
YEAH!

**Pull request merged**